packages/itwin/tree-widget/src/test/trees/models-tree/ModelsTreeComponent.test.tsx

```diff
@@ -39,8 +45,118 @@ describe("<ModelsTreeComponent />", () => {
+  } as unknown as Viewport;
+
+  it("returns null if iModel is undefined", () => {
+    sinon.stub(IModelApp.viewManager, "selectedView").get(() => ({} as Viewport));
+    const modelsTreeSpy = sinon.stub(modelsTree, "ModelsTree");
+    const result = render(
+      <ModelsTreeComponent />
+    );
+    expect(result.container.children).to.be.empty;
+    expect(modelsTreeSpy).to.not.be.called;
+  });
+
+  it("returns null if viewport is undefined", () => {
+    sinon.stub(UiFramework, "getIModelConnection").returns({} as IModelConnection);
+    const modelsTreeSpy = sinon.stub(modelsTree, "ModelsTree");
+    const result = render(
+      <ModelsTreeComponent />
+    );
+    expect(result.container.children).to.be.empty;
+    expect(modelsTreeSpy).to.not.be.called;
+  });
+
+  it("returns ModelsTreeComponentImpl when iModel and viewport are defined", async () => {
+    const modelsTreeSpy = sinon.stub(modelsTree, "ModelsTree").returns(<></>);
+    sinon.stub(IModelApp.viewManager, "selectedView").get(() => viewport);
+    sinon.stub(UiFramework, "getIModelConnection").returns({} as IModelConnection);
+    const result = render(
+      <ModelsTreeComponent />
+    );
+    expect(result.container.children).to.not.be.empty;
+    expect(modelsTreeSpy).to.be.called;
+  });
+
+  it("getLabel returns translated label of the component", () => {
+    const translateSpy = sinon.stub(TreeWidget, "translate").returns("test models label");
+    expect(ModelsTreeComponent.getLabel()).to.be.eq(TreeWidget.translate("test models label"));
+    expect(translateSpy).to.be.calledWith("models");
+  });
+
+  describe("queryModels", () => {
+
+    it("sets available models", async () => {
+      const iModel = {
+        models: {
+          queryProps: () => Promise.resolve([{
+            id: "testIdFromQueryModels",
+            modeledElement: {
+              id: "id",
+            },
+            classFullName: "className",
+          }]) as unknown as ModelProps[],
+        } as unknown as IModelConnection.Models,
+      } as unknown as IModelConnection;
+      const spy = sinon.stub().returns(<></>);
+      sinon.stub(modelsTree, "ModelsTree").returns(<></>);
+      sinon.stub(IModelApp.viewManager, "selectedView").get(() => viewport);
+      sinon.stub(UiFramework, "getIModelConnection").returns(iModel);
+      render(
+        <ModelsTreeComponent
+          headerButtons={[spy]}
+        />
+      );
+      await waitFor(() => expect(spy).to.be.calledWith(sinon.match((props: ModelsTreeHeaderButtonProps) => (props.models.length === 1 && props.models[0].id === "testIdFromQueryModels"))));
+    });
+
+    it("returns empty array when quryProps throws error", async () => {
+      const spy = sinon.stub().returns(<></>);
+      sinon.stub(modelsTree, "ModelsTree").returns(<></>);
+      sinon.stub(IModelApp.viewManager, "selectedView").get(() => viewport);
+      sinon.stub(UiFramework, "getIModelConnection").returns({} as IModelConnection);
```

Comment:
I think it would be better to make `queryProps` thrown error instead of using fake `IModelConnection` with undefined `queryProps`

Reply:
fixed